--- dropbear-2018.76/configure
+++ dropbear/configure
@@ -4743,11 +4743,13 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char deflate ();
+#include <zlib.h>
 int
 main ()
 {
-return deflate ();
+#ifndef ZLIB_VERSION
+#error
+#endif
   ;
   return 0;
 }
@@ -4797,11 +4799,13 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char deflate ();
+#include <zlib.h>
 int
 main ()
 {
-return deflate ();
+#ifndef ZLIB_VERSION                                             
+#error                                              
+#endif                       
   ;
   return 0;
 }
--- dropbear-2018.76/session.h
+++ dropbear/session.h
@@ -25,6 +25,10 @@
 #ifndef DROPBEAR_SESSION_H_
 #define DROPBEAR_SESSION_H_
 
+#ifndef z_streamp
+#define z_streamp mz_streamp
+#endif
+
 #include "includes.h"
 #include "buffer.h"
 #include "signkey.h"
