--- a/scripts/make.sh
+++ b/scripts/make.sh
@@ -283,7 +283,7 @@
   # Dump #define list for limits.h and unistd.h, create sed expression to
   # match known defines
   DEFINES="$(echo -e '#include <limits.h>\n#include <unistd.h>' | \
-    gcc -E -dM - | \
+    cc -E -dM - | \
     sed -n 's@^#define[ \t][ \t]*\([^ \t(]*\)[ \t].*@s/^\1$/\&/@p' )"
 
   # Extract limit names, compare against limits.h #defines, replace unknown
