--- miniz-2.0.7/CMakeLists.txt
+++ miniz/CMakeLists.txt
@@ -12,7 +12,7 @@
 
 set(miniz_SOURCE miniz.c miniz_zip.c miniz_tinfl.c miniz_tdef.c)
 
-add_library(miniz ${miniz_SOURCE})
+add_library(miniz SHARED ${miniz_SOURCE})
 
 set(EXAMPLE1_SRC_LIST "${CMAKE_CURRENT_SOURCE_DIR}/examples/example1.c")
 set(EXAMPLE2_SRC_LIST "${CMAKE_CURRENT_SOURCE_DIR}/examples/example2.c")
@@ -49,4 +49,25 @@
     LIBRARY DESTINATION lib
     )
 file(GLOB INSTALL_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/*.h)
-install(FILES ${INSTALL_HEADERS} DESTINATION include/${PROJECT_NAME})
\ No newline at end of file
+install(FILES ${INSTALL_HEADERS} DESTINATION include)
+
+macro(install_symlink filepath symname symdir)
+	install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${filepath} ${CMAKE_CURRENT_BINARY_DIR}/${symname})")
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${symname} DESTINATION ${symdir})
+endmacro(install_symlink)
+
+macro(install_empty name dir)
+	install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/${name})")
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${name} DESTINATION ${dir})
+endmacro(install_empty)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/miniz.pc.in ${CMAKE_BINARY_DIR}/miniz.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/miniz.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+install_symlink(miniz.pc zlib.pc ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+install_symlink(libminiz.so libz.so ${CMAKE_INSTALL_PREFIX}/lib)
+install_symlink(miniz.h zlib.h ${CMAKE_INSTALL_PREFIX}/include)
+install_empty(zconf.h ${CMAKE_INSTALL_PREFIX}/include)
--- miniz-2.0.7/miniz.pc.in
+++ miniz/miniz.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: miniz
+Description: Single C source file zlib-replacement library
+Version: @VERSION@
+Libs: -L${libdir} -lminiz
+Cflags: -I${includedir}
