--- libarchive-3.3.2/libarchive/archive_read_support_format_7zip.c
+++ libarchive/libarchive/archive_read_support_format_7zip.c
@@ -1211,9 +1211,10 @@
 #endif
 	case _7Z_DEFLATE:
 #ifdef HAVE_ZLIB_H
-		if (zip->stream_valid)
-			r = inflateReset(&(zip->stream));
-		else
+		if (zip->stream_valid) {
+			inflateReset(&(zip->stream));
+			r = Z_OK;
+		} else
 			r = inflateInit2(&(zip->stream),
 			    -15 /* Don't check for zlib header */);
 		if (r != Z_OK) {
--- libarchive-3.3.2/libarchive/archive_read_support_format_cab.c
+++ libarchive/libarchive/archive_read_support_format_cab.c
@@ -1433,9 +1433,10 @@
 		cab->stream.next_out = NULL;
 		cab->stream.avail_out = 0;
 		cab->stream.total_out = 0;
-		if (cab->stream_valid)
-			r = inflateReset(&cab->stream);
-		else
+		if (cab->stream_valid) {
+			inflateReset(&cab->stream);
+			r = Z_OK;
+		} else
 			r = inflateInit2(&cab->stream,
 			    -15 /* Don't check for zlib header */);
 		if (r != Z_OK) {
@@ -1577,13 +1578,14 @@
 	 */
 	if (cab->entry_cffolder->cfdata_index <
 	    cab->entry_cffolder->cfdata_count) {
-		r = inflateReset(&cab->stream);
+		inflateReset(&cab->stream);
+		r = Z_OK;
 		if (r != Z_OK)
 			goto zlibfailed;
-		r = inflateSetDictionary(&cab->stream,
+		/*r = inflateSetDictionary(&cab->stream,
 		    cab->uncompressed_buffer, cfdata->uncompressed_size);
 		if (r != Z_OK)
-			goto zlibfailed;
+			goto zlibfailed;*/
 	}
 
 	d = cab->uncompressed_buffer + cfdata->read_offset;
--- libarchive-3.3.2/libarchive/archive_read_support_format_iso9660.c
+++ libarchive/libarchive/archive_read_support_format_iso9660.c
@@ -1567,9 +1567,10 @@
 		zisofs->block_off += 4;
 
 		/* Initialize compression library for new block. */
-		if (zisofs->stream_valid)
-			r = inflateReset(&zisofs->stream);
-		else
+		if (zisofs->stream_valid) {
+			inflateReset(&zisofs->stream);
+			r = Z_OK;
+		} else
 			r = inflateInit(&zisofs->stream);
 		if (r != Z_OK) {
 			archive_set_error(&a->archive, ARCHIVE_ERRNO_MISC,
--- libarchive-3.3.2/libarchive/archive_read_support_format_zip.c
+++ libarchive/libarchive/archive_read_support_format_zip.c
@@ -1274,9 +1274,10 @@
 
 	/* If we haven't yet read any data, initialize the decompressor. */
 	if (!zip->decompress_init) {
-		if (zip->stream_valid)
-			r = inflateReset(&zip->stream);
-		else
+		if (zip->stream_valid) {
+			inflateReset(&zip->stream);
+			r = Z_OK;
+		} else
 			r = inflateInit2(&zip->stream,
 			    -15 /* Don't check for zlib header */);
 		if (r != Z_OK) {
--- libarchive-3.3.2/libarchive/archive_write_set_format_iso9660.c
+++ libarchive/libarchive/archive_write_set_format_iso9660.c
@@ -7950,9 +7950,10 @@
 		zisofs->block_off += 4;
 
 		/* Initialize compression library for new block. */
-		if (zisofs->stream_valid)
-			r = inflateReset(&zisofs->stream);
-		else
+		if (zisofs->stream_valid) {
+			inflateReset(&zisofs->stream);
+			r = Z_OK;
+		} else
 			r = inflateInit(&zisofs->stream);
 		if (r != Z_OK) {
 			archive_set_error(&a->archive, ARCHIVE_ERRNO_MISC,
